core/src/View/Kokkos_ViewCtor.hpp

```diff
@@ -213,11 +213,16 @@ struct ViewCtorProp : public ViewCtorProp<void, P>... {
   using execution_space = typename var_execution_space::type;
   using pointer_type    = typename var_pointer::type;
 
-  /*  Copy from a matching argument list.
-   *  Requires  std::is_same< P , ViewCtorProp< void , Args >::value ...
-   */
-  template <typename... Args>
-  inline ViewCtorProp(Args const &...args) : ViewCtorProp<void, P>(args)... {}
+  // Construct from a matching argument list.
+  //
+  // Note that if P is empty, this constructor is the default constructor.
+  // On the other hand, if P is not empty, the constraint implies that
+  // there is no default constructor.
+  template <typename... Args,
+            typename = std::enable_if_t<std::conjunction_v<
```

Comment:
I don't quite understand this.

`conjunction` with just one argument is unusual.

And the comment doesn't see right, either.  Whether or not this acts as a default constructor depends not only on the constraint but also on `sizeof...(Args) == 0`.

Reply:
The arguments are the expansion of this: `std::is_constructible<view_ctor_prop_base<P>, Args &&>...`